packages/cli/test/fixtures/discover/mem.datasource.js.txt

```diff
@@ -0,0 +1,29 @@
+"use strict";
+var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
```

Comment:
Hmm, keeping versioning JS files is not pretty. It makes me wonder how are tests for other generators (e.g. `lb4 repository`) dealing with datasource discovery? Could you PTAL and see if there is a way how to use an existing approach? Do we actually need this `.js` file? Isn't it enough to have datasource `.json` file?

Reply:
You're right, I'm going to try and do this with the `legacy.Datasource` constructor and a JSON file.  The only downside I can see is that if any config is defined in`server/datasources/datasource.ts` it won't be loaded properly

Reply:
I've tried this, but if I only use a .json file it will crash because the `lb4` command doesn't have the connector.  Is it worth it to try and bootstrap the connectors found in the discovering `node_modules`?

Reply:
Could you please create a new feature branch in your repo where you will commit the code showing what have you tried, so that I can run that version myself and better understand what's going on under the hood? I am afraid I am not able to build a good picture in my head just from our discussion.

Reply:
### Compare loading from `name.datasource.json`, to loading `name.datasource.js`

(requires lb4 project with non-memory datasource in `dist`, it doesn't have to connect successfully)

```
git clone https://github.com/marvinirwin/loopback-next --single-branch --branch discovering-from-json  .;
cd packages/cli;
npm run build;
npm link;
cd SOME_LB4_PROJECT;
//  new DataSource(JSON.parse(fs.readFileSync(pathToJson).toString()))
// This should tell you to npm install the connector
LOAD_JSON_DATASOURCES=true lb4 discover;
// new require(pathToJavascript);
// This will not tell you to npm install the connector
LOAD_JSON_DATASOURCES= lb4 discover;
```